backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/CodeGenerator.kt

```diff
@@ -85,18 +87,19 @@ internal class CodeGenerator(override val context: Context) : ContextUtils {
     }
 
     fun typeInfoForAllocation(constructedClass: ClassDescriptor): LLVMValueRef {
-        val descriptorForTypeInfo = if (constructedClass.isObjCClass()) {
-            context.ir.symbols.objCPointerHolder.owner
-        } else {
-            constructedClass
-        }
-        return typeInfoValue(descriptorForTypeInfo)
+        assert(!constructedClass.isObjCClass())
+        return typeInfoValue(constructedClass)
     }
 
     fun generateLocationInfo(locationInfo: LocationInfo): DILocationRef? {
         return LLVMCreateLocation(LLVMGetModuleContext(context.llvmModule), locationInfo.line, locationInfo.line, locationInfo.scope)
     }
 
+    val objCDataGenerator = when (context.config.target) {
+        KonanTarget.IOS_ARM64, KonanTarget.IOS_X64, KonanTarget.MACOS_X64 -> ObjCDataGenerator(this)
+        else -> null
```

Comment:
Why do you return null instead of throwing an exception, e.g. IllegalStateException with description about incorrect target

Reply:
Because this code works for all targets, and having `null` on target not supporting Objective-C interop is the intentional behaviour here.